coding/string_utf8_multilang.hpp

```diff
@@ -129,6 +129,24 @@ class StringUtf8Multilang
     }
   }
 
+  template <typename Fn>
+  void ForEachOf(std::vector<std::string> languages, Fn && fn) const
```

Comment:
1. s/ForEachOf/ForEachLanguage/ ?
2. Please document that the order of |languages| is important.
3. Overall, this looks like a bad idea when the list of languages is long. Note that GetString is not O(1).
A better way would be to use the regular ForEach and filter out the languages you don't want (or save the best language and then call GetString once).

Reply:
1, 2 Fixed.
3. I need to check transliteration of that languages in specified order, and stop at good enough.
So I can index all translations at construction time, but it can affect other users.
Or I can use regular ForEach, store results and then search in the results, but it is not going to be faster